RevenueCatUI/Templates/Example1Template.swift

```diff
@@ -173,8 +178,12 @@ private struct Example1TemplateContent: View {
 
     @ViewBuilder
     private var button: some View {
-        Button {
+        AsyncButton {
+            let cancelled = try await self.purchaseHandler.purchase(package: self.data.package).userCancelled
 
+            if !cancelled {
+                await self.dismiss()
```

Comment:
This should work regardless of how users display `PaywallView`:
Per the docs, it can:
- Dismiss a modal presentation, like a sheet or a popover.
- Pop the current view from a `NavigationStack
- Close a window that you create with `WindowGroup`

Reply:
Good point, especially from what we talked about today.
Especially for that I think we should add a sort of "event handler" / SwiftUI property like StoreKit views.